### ISSUE TITLE
md-chips with md-autocomplete[md-require-match="true"] and [md-add-on-blur]

### ISSUE THREAD

**Issue by reppners** [angular/material#9682]
**Actual Behavior**:
When using md-chips with md-autocomplete and md-require-match being true it should not be possible to add new labels. When md-add-on-blur is also true chips get added on blur, though the require match prohibits chip adding.

**Expected Behavior**:
When require-match is used the chip should not be added on blur.

**Angular Versions**:
 -  `Angular Version: 1.5.8`
 -  `Angular Material Version: 1.1.1`

**Comment by DevVersion** [comment]
Duplicate of #9582.

**Comment by reppners** [comment]
Workaround which fixes this issue but not #9582 

```js
angular.module( "mdChipsOverride", [] ).directive( "mdChips", mdChips );

function mdChips()
{
    return {
        restrict: "E",
        require: "mdChips", // Extends the original mdChips directive
        link: function ( _scope:IScope, _element:IAugmentedJQuery, _attributes:IAttributes, mdChipsCtrl:any )
        {
            mdChipsCtrl.onInputBlur = function ( this:any, _event:JQueryEventObject )
            {
                this.inputHasFocus = false;

                let chipBuffer = this.getChipBuffer().trim();

                // Update the custom chip validators.
                this.validateModel();

                let isModelValid = this.ngModelCtrl.$valid;

                if( this.userInputNgModelCtrl )
                {
                    isModelValid &= this.userInputNgModelCtrl.$valid;
                }

                // Only append the chip and reset the chip buffer if the chips and input ngModel is valid.
                if( this.addOnBlur && chipBuffer && isModelValid && !this.requireMatch )
                {
                    this.appendChip( chipBuffer );
                    this.resetChipBuffer();
                }
            };
        }
    };
}
```
